src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  dead = false;
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore &&
+      animal.hidden === false &&
+      animal.dead !== true
```

Comment:
[CODE REDUNDANCY] The check for 'animal.dead !== true' is not needed. According to the task, the health property determines if the animal is alive. If health is less than or equal to 0, the animal should be considered dead and removed from the 'Animal.alive' array.